Include/cpython/object.h

```diff
@@ -167,10 +167,23 @@ typedef struct {
     objobjargproc mp_ass_subscript;
 } PyMappingMethods;
 
+#if !defined(Py_LIMITED_API) || Py_LIMITED_API+0 >= 0x030A0000
```

Comment:
`cpython/object.h` is only imported when `Py_LIMITED_API` is not defined, so this `#if` does not make sense. Also, moving the declaration of `PySendResult` here removes it from the limited API. It actually break the limited API because importing `abstract.h` with the limited API will be error.